src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+        return dtos.stream()
+                .filter(dto -> dto.hasNotes() == hasNotes)
+                .collect(Collectors.toList());
+    }
+
+    private void buildDetailSections(List<JaggaerUploadSupplierXmlFileDetailsDto> xmlFileDtos) {
+        xmlFileDtos.stream().forEach(dto -> {
+            reportWriterService.writeNewLines(4);
+            reportWriterService.writeSubTitle(formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_SECTION_HEADER, dto.getXmlFileName()));
+            buildDetailLines(filterDetailDto(dto.getActiveVendors(), true));
+            buildDetailLines(filterDetailDto(dto.getActiveVendors(), false));
+            buildDetailLines(filterDetailDto(dto.getInactiveVendors(), true));
+            buildDetailLines(filterDetailDto(dto.getInactiveVendors(), false));
+            reportWriterService.writeSubTitle(formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_SECTION_FOOTER));
+        });
+    }
+
+    private void buildDetailLines(List<JaggaerUploadSupplierVendorDetailDto> details) {
+        details.stream().forEach(detail -> {
+            reportWriterService.writeFormattedMessageLine(formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_DETAIL_LINE,
+                        detail.isActive() ? ACTIVE : INACTIVE, detail.getVendorName(), detail.getVendorNumber()));
+            
+            detail.getNotes().stream().forEach(note -> 
+                reportWriterService.writeFormattedMessageLine(
+                        formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_DETAIL_NOTE_LINE, note)));
+        });
+    }
+    
+    private void emailReport() {
+        final String toAddress = findReportToAddress();
+        final String fromAddress = toAddress;
+        final String body = LoadFileUtils.safelyLoadFileString(reportWriterService.getReportFile().getAbsolutePath());
+        
+        if (StringUtils.isNotBlank(toAddress)) {        
+            BodyMailMessage message = new BodyMailMessage();
+            message.setFromAddress(fromAddress);
+            String subject = reportWriterService.getTitle();
+            message.setSubject(subject);
+            message.getToAddresses().add(toAddress);
+            message.setMessage(body);
+    
+            boolean htmlMessage = false;
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("sendEmail, from address: " + fromAddress + "  to address: " + toAddress);
+                LOG.debug("sendEmail, the email subject: " + subject);
+                LOG.debug("sendEmail, the email budy: " + body);
+            }
+            try {
+                emailService.sendMessage(message, htmlMessage);
+            } catch (Exception e) {
+                LOG.error("sendEmail, the email could not be sent", e);
+            }
+        } else {
+            LOG.info("emailReport, the JAGGAER_XML_REPORT_EMAIL parameter is empty, so not emailing the report");
```

Comment:
Is logging this message at info level okay, or should it be logged at warning or error level instead?

Reply:
good point, I will switch to warn